pyomo/core/expr/numeric_expr.py

```diff
@@ -784,6 +785,45 @@ class MonomialTermExpression(ProductExpression):
     def getname(self, *args, **kwds):
         return 'mon'
 
+class DivisionExpression(ExpressionBase):
+    """
+    Division expressions::
+
+        x/y
+    """
+    __slots__ = ()
+    PRECEDENCE = 4
+
+    def nargs(self):
+        return 2
+
+    def _precedence(self):
+        return DivisionExpression.PRECEDENCE
+
```

Comment:
I suspect that _cpmpute_polynomial_degree is not returning the correct polynomial degree for many expressions.

1) Can you clarify why we only return a result if the denominator is constant? Is it just that it is too difficult to compute this for more complex expressions?

2) Is other code assuming that None means "not known"?

Reply:
`None` means "not polynomial".  Division expressions with non-constant denominators are not polynomial.